specifyweb/specify/views.py

```diff
@@ -91,6 +92,18 @@ def raise_error(request):
     raise Exception('This error is a test. You may now return to your regularly '
                     'scheduled hacking.')
 
+def filter_rank_deletion_exception(obj, delete_blockers):
+    # Check if the object is a tree rank
+    if not is_instance_of_tree_def_item(obj):
+        return
+    # Filter out blocker that is the child tree rank of tree rank being deleted.
+    # This is handled by the business rules when deleting a tree rank, the parent is set to the grandparent.
+    delete_blockers[:] = list(
+        filter(
+            lambda db: db['field'] != 'parent' and db['table'] == type(obj).__name__, 
+            delete_blockers
+        )
+    )
```

Comment:
Ya, I agree that it should be done on the front-end.  I just put this in as an alternative solution for Caroline.  Not going to keep this code after we implement the front-end solution.  See my earlier comment:

 > Ya, that error should be occurring, because there is a taxon in the 'Test' Rank. The Rank needs to be empty of taxon before being deleted. We should add a check on the front-end to make sure this is the case before calling the delete api. Ya, that example for the delete blocker pop-up would be a good idea in this instance.